numpy/lib/index_tricks.py

```diff
@@ -589,6 +590,336 @@ def __iter__(self):
     next = __next__
 
 
+def fproduct(*iterables, **kwargs):
+    """
+    Like itertools.product, but in fortran order
+    """
+    return (i[::-1] for i in itertools.product(*iterables[::-1], **kwargs))
+
+
+class ndrange(_nx.collections_abc.Sequence):
+    """
+    An N-dimensional range object that returns an iterator (or reversed
+    iterator) tthat can produce a sequence of tuples from
+    start (inclusive tuple) to stop (exclusive tuple) by step.
```

Comment:
Some of this isn't quite accurate. It doesn't return an iterator, rather, it can be iterated.

What about more closely adapting the `range` docstring?

Reply:
I tried. Let me know what you think.